src/Compilers/CSharp/Portable/Symbols/Metadata/PE/TupleTypeDecoder.cs

```diff
@@ -60,13 +60,15 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE
     /// </summary>
     internal struct TupleTypeDecoder
     {
+        private readonly bool _nonNullTypes;
```

Comment:
Why does tuple type decoding need to account for NonNullTypes at all?
This decoder adds tuple element names.

It feels like we should just carry through the NonNullTypes context from a type to the decoded type.

Reply:
Good catch, thanks.

---
In reply to: [203881061](https://github.com/dotnet/roslyn/pull/28687#discussion_r203881061) [](ancestors = 203881061)